src/main/java/com/microsoft/azure/util/AzureCredentials.java

```diff
@@ -140,7 +140,7 @@ public ServicePrincipal(
             this.clientSecret = Secret.fromString(clientSecret);
             this.oauth2TokenEndpoint = Secret.fromString(oauth2TokenEndpoint);
             this.tenant = Secret.fromString(
-                    ServicePrincipal.getTenantFromTokenEndpoint(oauth2TokenEndpoint));
+                    ServicePrincipal.getTenantFromTokenEndpoint(this.oauth2TokenEndpoint.getPlainText()));
```

Comment:
What's the difference? Shouldn't `this.oauth2TokenEndpoint.getPlainText()` equals to `oauth2TokenEndpoint`